### PR TITLE
Renew plan

### PR DESCRIPTION
**Pull request by proteusvacuum**
http://manage.dimagi.com/default.asp?167936

This allows users to change their subscription type when renewing a subscription.

renew_subscription

Code buddy @orangejenny 
Context: @nickpell @benrudolph

### CODE REVIEW

File: corehq/apps/accounting/tests/test_renew_subscription.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import datetime
+from corehq import Domain
+from corehq.apps.accounting.tests.base_tests import BaseAccountingTest
+from corehq.apps.accounting import generator
+from corehq.apps.accounting.models import (
+        Subscription,
```
Comment (dimagimon) [inline corehq/apps/accounting/tests/test_renew_subscription.py]: E126 continuation line over-indented for hanging indent

File: corehq/apps/accounting/tests/test_renew_subscription.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        BillingAccount,
+        DefaultProductPlan,
+        SoftwarePlanEdition
+    )
```
Comment (dimagimon) [inline corehq/apps/accounting/tests/test_renew_subscription.py]: E121 continuation line under-indented for hanging indent

**Comment by orangejenny** [comment]
Looks reasonable to me, will let someone more familiar with billing merge.

File: corehq/apps/accounting/tests/test_renew_subscription.py (hunk 1 above)
Comment (nickpell) [inline corehq/apps/accounting/tests/test_renew_subscription.py]: would be good to fix, but not blocking merge

File: corehq/apps/accounting/tests/test_renew_subscription.py (hunk 2 above)
Comment (nickpell) [inline corehq/apps/accounting/tests/test_renew_subscription.py]: ditto

**Comment by nickpell** [comment]
one blocking concern, then should be good to go.  This is awesome :)

**Comment by nickpell** [comment]
does this need QA?  @christineawhite

**Comment by christineawhite** [comment]
@nickpell can't hurt right?

**Comment by nickpell** [comment]
agree

**Comment by proteusvacuum** [comment]
this passed QA, so should be g2g

**Pull request merged**